### PR TITLE
docs(v2): add documentation about importing code files

### LINKED ISSUE #4540: Include code snippets
## 🚀 Feature

I'm not sure if you can do this currently, but it would be nice if there was a way to include code snippets from a file:
```html
You can do X like so:

{{ include "foo.js" }}

// expands to

'''js
// the code from foo.js
'''
```

## Motivation

Many other static site generators include this feature, and having it would make migration easier. It would also allow you to test/compile those examples in CI to make sure the code is up to date.

Is this possible currently with react components?

### PR DESCRIPTION
**Pull request by Josh-Cena**
## Motivation

Resolve #4540. Yet another PR trying to migrate contents from the example MD page to the real documentation. I wonder what's the ["recipes section"](https://github.com/facebook/docusaurus/issues/4540#issuecomment-811391802) going to look like and whether I can help with that 🤔

I also did some minor editing to my previously written section. (#4958)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### CODE REVIEW

**Comment by slorber** [comment]
Thanks

I'm going to merge this because some users already use it and we also use it internally, but some concerns:
- raw loader is deprecated and we'll provide a better way to obtain the string of a source file (part of https://github.com/facebook/docusaurus/pull/4708)
- `className="language-jsx"` is a weird API we currently have for code blocks, I'd like it to be refactored

File: website/docs/guides/markdown-features/markdown-features-react.mdx [hunk 1]
```diff
@@ -72,6 +72,34 @@ Since all doc files are parsed using MDX, any HTML is treated as JSX. Therefore,
...
+
+</BrowserWindow>
+
+<br />
+
+:::note
+
+You have to use `<CodeBlock>` rather than the Markdown triple-backtick ` ``` `, because the latter will ship out any of its content as-is, but you want JSX to insert the imported text here.
+
+:::
```
Comment (slorber) [inline website/docs/guides/markdown-features/markdown-features-react.mdx:101]: can you add a warning here and mention that this is experimental and API breaking changes are expected on this

we'll replace raw loader by https://webpack.js.org/guides/asset-modules/#replacing-inline-loader-syntax

Comment (Josh-Cena) [reply website/docs/guides/markdown-features/markdown-features-react.mdx:101]: Sure, will work on this in a moment

Comment (Josh-Cena) [reply website/docs/guides/markdown-features/markdown-features-react.mdx:101]: > we'll replace raw loader by https://webpack.js.org/guides/asset-modules/#replacing-inline-loader-syntax

Should I change my demo case in the documentation, or is this just a plan?

Comment (slorber) [reply website/docs/guides/markdown-features/markdown-features-react.mdx:101]: we don't support `?raw` syntax yet so we'll document it once we support it, for now you can keep documenting `raw-loader` usage

Comment (Josh-Cena) [reply website/docs/guides/markdown-features/markdown-features-react.mdx:101]: Should I add a warning then, or should we just keep it in mind and come back once the support is in place?

File: website/docs/guides/markdown-features/markdown-features-react.mdx [hunk 2]
```diff
@@ -100,6 +100,12 @@ You have to use `<CodeBlock>` rather than the Markdown triple-backtick ` ``` `,
 
 :::
 
+:::warning
+
+The `className="language-jsx"` prop is experimental and might be subject to breaking API changes in the future.
```
Comment (slorber) [inline website/docs/guides/markdown-features/markdown-features-react.mdx:105]:
```suggestion
This feature is experimental and might be subject to API breaking changes in the future.
```

**Review by slorber — APPROVED** [review]

**Pull request merged**